tests/unit/test_local_data.py

```diff
@@ -0,0 +1,242 @@
+@patch('sagemaker.utils.download_folder')
+@patch('tempfile.mkdtemp', lambda dir: '/tmp/working_dir')
+def test_s3_data_source(download_folder, LocalFileDataSource, sagemaker_local_session):
+    data_source = sagemaker.local.data.S3DataSource('my_bucket', '/transform/data', sagemaker_local_session)
+    download_folder.assert_called()
+    data_source.get_file_list()
+    LocalFileDataSource().get_file_list.assert_called()
+    data_source.get_root_dir()
+    LocalFileDataSource().get_root_dir.assert_called()
+
+
+def test_get_splitter_instance_with_valid_types():
+    splitter = sagemaker.local.data.get_splitter_instance(None)
+    assert isinstance(splitter, sagemaker.local.data.NoneSplitter)
+
+    splitter = sagemaker.local.data.get_splitter_instance('Line')
+    assert isinstance(splitter, sagemaker.local.data.LineSplitter)
+
+    splitter = sagemaker.local.data.get_splitter_instance('RecordIO')
+    assert isinstance(splitter, sagemaker.local.data.RecordIOSplitter)
+
+
+def test_get_splitter_instance_with_invalid_types():
+    with pytest.raises(ValueError):
+        # something invalid
+        sagemaker.local.data.get_splitter_instance('JSON')
+
+
+def test_none_splitter(tmpdir):
+    test_file_path = tmpdir.join('none_test.txt')
+
+    with test_file_path.open('w') as f:
+        f.write('this\nis\na\ntest')
+
+    splitter = sagemaker.local.data.NoneSplitter()
+    data = [x for x in splitter.split(str(test_file_path))]
+    assert data == ['this\nis\na\ntest']
+
+
+def test_line_splitter(tmpdir):
+    test_file_path = tmpdir.join('line_test.txt')
+
+    with test_file_path.open('w') as f:
+        for i in range(10):
+            f.write('%s\n' % i)
+
+    splitter = sagemaker.local.data.LineSplitter()
+    data = [x for x in splitter.split(str(test_file_path))]
+    assert len(data) == 10
+    for i in range(10):
+        assert data[i] == '%s\n' % str(i)
+
+
+def test_recordio_splitter(tmpdir):
+
+    test_file_path = tmpdir.join('recordio_test.txt')
+    with test_file_path.open('wb') as f:
+        for i in range(10):
+            data = str(i).encode('utf-8')
+            sagemaker.amazon.common._write_recordio(f, data)
+
+    splitter = sagemaker.local.data.RecordIOSplitter()
+    data = [x for x in splitter.split(str(test_file_path))]
+
+    assert len(data) == 10
+
+
+def test_get_batch_strategy_instance_with_valid_type():
+    # Single Record
+    strategy = sagemaker.local.data.get_batch_strategy_instance('SingleRecord', None)
+    assert isinstance(strategy, sagemaker.local.data.SingleRecordStrategy)
+
+    # Multi Record
+    strategy = sagemaker.local.data.get_batch_strategy_instance('MultiRecord', None)
+    assert isinstance(strategy, sagemaker.local.data.MultiRecordStrategy)
+
+
+def test_get_batch_strategy_instance_with_invalid_type():
+    with pytest.raises(ValueError):
+        # something invalid
+        sagemaker.local.data.get_batch_strategy_instance('NiceRecord', None)
```

Comment:
personal preference - I'd just make the string 'SomethingInvalid' instead of writing a comment to explain it